### ISSUE TITLE
[Question] Getting poor performance when loading trained PPO agent

### ISSUE THREAD

**Issue by ScottHerod-AMRC** [DLR-RM/stable-baselines3#2113]
### ❓ Question

Im training a PPO agent on a custom environment, when I save a model that reaches a certain performance plateau and reload it, the agent just takes the same action over and over again. 

I am using a VecNormalize wrapper, but as far as I know, I am loading and saving the normalisation data as specified in the docs. Is there anything else I need to account for to reproduce the training behaviour after loading?

Here is my training code:
```Python
training_steps = 1_000_000
update_interval = 500

vec_env = make_vec_env(lambda: TimeLimit(PlantsimEnvironment(), update_interval))

vec_env = VecNormalize(vec_env, norm_obs=True, norm_reward=True)

callbacks = CallbackList([SaveBestModelCallback("saved_models/PPO"),
                          LogAndPlotCallback("logs/PPO/log_PPO_1.csv"),
                          LearningRateDecayCallback(1e-4, training_steps)])

model = PPO("MlpPolicy", vec_env,
            verbose=1,
            device="cpu",
            n_steps=update_interval,
            n_epochs=20,
            learning_rate=1e-3,
            batch_size=100,
            tensorboard_log="logs/PPO/",
            policy_kwargs=dict(net_arch=[256, 256])
            )

model.learn(training_steps,
            progress_bar=True,
            callback=callbacks,
            tb_log_name="PPO")
```

I am saving the model and normalisation data in a custom callback which is triggering the save on `_on_rollout_end`

```Python
    def _on_rollout_end(self) -> None:
        # check throughput
        if self.bridge.throughput > self.best_throughput:
            self.best_throughput = self.bridge.throughput
            print("New best model, saving model!")
            self.model.save(f"{self.log_dir}/best_model_{self.num_timesteps}S_{self.best_throughput}T")
            if isinstance(self.model.get_env(), VecNormalize):
                self.model.get_env().save(f"{self.log_dir}/best_model_{self.num_timesteps}S_{self.best_throughput}T.pkl")

```

```Python
run_length = 2000

model_path = "saved_models/PPO2/best_model_139000S_22T"
normalisation_data = model_path + ".pkl"

vec_env = make_vec_env(lambda: TimeLimit(PlantsimEnvironment(), run_length))
vec_env = VecNormalize.load(normalisation_data, vec_env)

vec_env.training = False
vec_env.norm_reward = False

model = PPO.load(model_path, env=vec_env, device="cpu")

obs = vec_env.reset()

for _ in range(run_length):
    action, states = model.predict(obs)
    print(action)
    obs, rewards, _, info = vec_env.step(action)
```

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [x] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [x] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by ScottHerod-AMRC** [comment]
I think this issue may stem from some of my models suffering from catastrophic forgetting. Upon tweaking my model params and getting a few more models trained, they at least don't seem to be just taking the same action repeatedly. This leads me to believe that I am at least saving the normalisation data correctly.

Even though my `LogAndPlotCallback` and `SaveBestModelCallback` are both triggered on `_on_rollout_end` some of the saved models don't perform as well as they should when loaded. I'm still not sure why, the rollout end is triggered before the model is updated so the performance statistics should be directly related to the model being saved.

The performance isn't utterly terrible so the stochastic nature of the environment and training might explain the discrepancy somewhat. I don't think there is a bug going on here, probably just poorly trained models on my side, so I'll close this issue.
